backend/src/main/java/com/redhat/cloud/notifications/db/repositories/EndpointRepository.java

```diff
@@ -0,0 +1,215 @@
+package com.redhat.cloud.notifications.db.repositories;
+
+import com.redhat.cloud.notifications.db.Query;
+import com.redhat.cloud.notifications.db.session.CommonStateSessionFactory;
+import com.redhat.cloud.notifications.models.CamelProperties;
+import com.redhat.cloud.notifications.models.EmailSubscriptionProperties;
+import com.redhat.cloud.notifications.models.Endpoint;
+import com.redhat.cloud.notifications.models.EndpointProperties;
+import com.redhat.cloud.notifications.models.EndpointType;
+import com.redhat.cloud.notifications.models.EventType;
+import com.redhat.cloud.notifications.models.WebhookProperties;
+import io.smallrye.mutiny.Uni;
+import org.hibernate.reactive.mutiny.Mutiny;
+import org.jboss.logging.Logger;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+// TODO: Move this class to notifications-engine.
+
+@ApplicationScoped
+public class EndpointRepository {
+
+    private static final Logger LOGGER = Logger.getLogger(EndpointRepository.class);
+
+    @Inject
+    Mutiny.SessionFactory sessionFactory;
+
+    @Inject
+    CommonStateSessionFactory commonStateSessionFactory;
+
+    public Uni<Endpoint> createEndpoint(Endpoint endpoint, boolean useStatelessSession) {
```

Comment:
This now exists in both `EndpointResources` and `EndpointRepository` (same code). `EndpointRepository` will be refactored later to make the code simpler.

Reply:
Is there any strategy to avoid repeating the code? I'm a bit worried that we might end up with different "versions" of the same code due to fixes in one and not the other.

Reply:
If and only if I don't have another choice, I will put it in the common module (which does not exist yet).